### ISSUE TITLE
livestream_saver does not handle age-restricted videos correctly

### ISSUE THREAD

**Issue by mechalincoln** [glubsy/livestream_saver#49]
Hey. This app worked for a long time, but the streamer I watch started to change his streams to age-restricted, and now livestream_saver no longer works.

I have included a gif showing what happens, even after providing correct login credentials.
![livestream_saver_issue](https://user-images.githubusercontent.com/49933679/197853074-cc504af4-acaa-424a-94aa-6d9466c919fb.gif)

Please look into why livestream_saver is not handling age restricted streams.

Thank you.

**Comment by glubsy** [comment]
Thanks for reporting. 

I believe there is no other way than providing cookies of a logged-in session in order to work around this issue. 

Non logged-in users are most likely blocked from accessing age restricted videos.

**Comment by mechalincoln** [comment]
I did that, and this is what still happens.

**Comment by mechalincoln** [comment]
Even with a formatted Netscape cookies file, it will do this.

**Comment by glubsy** [comment]
Can you share the channel Identifier or at least its name so that test that out?

**Comment by mechalincoln** [comment]
Thank you.

The channel ID is UCeajjc71kLsu9hVDkFvRHjg

He streams randomly, which is why I need this service.

FYI it's prank calls and lots of cussing, so not the most important thing, but something fun.

**Comment by glubsy** [comment]
Can you confirm you can view the live streams with your logged-in account? Perhaps your account is blocked for whatever reason. 

I'll monitor that channel for a while and keep an eye for that error.

**Comment by mechalincoln** [comment]
Yes, I can confirm that I can view them with my logged in account.

Thank you again.

**Comment by mechalincoln** [comment]
I think he's on now but age restriction is not on, so the script is currently working. If it does come on I'll let you know.

**Comment by mechalincoln** [comment]
He made that video age-restricted now.

The script is erroring out like in the gif now.

**Comment by glubsy** [comment]
Indeed I managed to reproduce the bug. I got 2 hours of the last stream (id was `kwVwhylnMsk`) and I assume it was not over when the recording stopped.

That will require some investigation and some work. 
For now what I suggest is trying to record with `yt-dlp` and see if that solves your problem (they might have implemented a workaround for the age restriction). You can spawn `yt-dlp` automatically if you follow the README.md carefully.

**Comment by mechalincoln** [comment]
Understood.

Thank you again for investigating. 😃

**Comment by mechalincoln** [comment]
On a somewhat unrelated note, livestream_saver is not currently detecting any videos either for that or another channel I watch, both of which have several videos.
![livestream_saver102822](https://user-images.githubusercontent.com/49933679/198753061-8c6fdab2-3e45-4dc2-9ea4-3600c3dc4e11.PNG)

I know that youtube has been screwing around with their interface lately, so that may be why.

**Comment by mechalincoln** [comment]
It's currently showing zero videos for both, when I know there are many more than zero videos for each.

**Comment by glubsy** [comment]
This is a separate issue that has been reported in #50 already.
